portfolio/src/main/webapp/script.js

```diff
@@ -14,15 +14,34 @@
 
 "use strict"
 
-// loads comments
+// is called when page loads
 document.addEventListener("DOMContentLoaded", () => {
-    fetch("/comments").then((response) => (response.json())).then((json) => {
+    // set max-comments variable and change event on it
+    const DEFAULT_COMMENTS_NUMBER = 3;
+    let maxComments = DEFAULT_COMMENTS_NUMBER;
+    const maxCommentsElement = document.getElementById("max-comments");
+    maxCommentsElement.addEventListener("change", (event) => {
+        maxComments = parseInt(event.target.value);
+        // if parseInt couldn't parse the value or the value is negative
+        if (isNaN(maxComments) || maxComments < 0) {
+            maxComments = DEFAULT_COMMENTS_NUMBER;
```

Comment:
In my opinion it is not a good idea to check this on both back-end and front-end because if you want to edit DEFAULT_COMMENTS_NUMBER later you have to edit it in 2 places and you may forget about the other.

I would remove DEFAULT_COMMENTS_NUMBER on front-end and if the user doesn't enter a valid maxComments value I would make an alert or something.

Reply:
Okay, thanks! I think it will be better to substitute DEFAULT_COMMENTS_NUMBER with MAX_COMMENTS_NUMBER on backend (so that default number stays only on frontend) and leave validation on whether maxComments is a number and a correct number on both frontend and backend. On frontend it is needed to make an alert and reduce server load, and on backend it is just must have.

Reply:
That sounds very good to me!